src/mbgl/text/glyph.hpp

```diff
@@ -55,17 +55,21 @@ class Glyph {
 };
 
 using Glyphs = std::map<GlyphID, optional<Immutable<Glyph>>>;
-using GlyphMap = std::map<FontStack, Glyphs>;
+using GlyphMap = std::map<FontStackHash, Glyphs>;
```

Comment:
Curious, why change this to a hash instead of using `std::unordered_map` with a custom `Hash` template argument?

Reply:
It's true, the explicit hashing does feel awkward. I think the part that feels worst is when we have to do the reverse from hash->[string] for dependency lookup (i.e. `GeometryTileWorker::onGlyphsAvailable`).

I'm interested in the "store an iterator" idea, but I'm concerned that it will make the code harder to audit because you have to make sure that if you store an iterator you're using the same map when you try to use the iterator later -- and GeometryTileWorker already merges GlyphMaps from multiple requests to GlyphManager over time...